### DISCUSSION TITLE
API question...

### DISCUSSION THREAD

**Discussion by adanecito** [Safecast/safecastapi#978] [Q&A] [answered]
Is there a rest API and docs for it? I want to retrieve a set of data points in json covering anywhere from a day to maybe a week.

Many Thanks!
-Tony

**Comment by robouden** [comment]
Tony,

The API is on api.safecast.org.
If you need help, please post here.

Regards,
rob Oudendijk

**Reply by adanecito** [reply]
There is no discussion groups or any way to post questions there. Thanks, I will not post anything here.
-Thanks,
Tony

**Comment by adanecito** [comment]
Ok, I went to the wiki that in turn lead me to the api page. But the page results when I tried let me have only 25 json records. How do I get more? Seems like there should be more. When I increase the date range I still only get 25 json records. Below is my example.

https://api.safecast.org/measurements.json?since=2022-09-08+00%3A00&until=2022-09-09+00%3A00

Thanks for any help!
Tony Anecito

**Comment by robouden** [comment, marked as answer]
How about [https://api.safecast.org/measurements.json?since=2022-09-08+00%3A00&until=2022-09-09+00%3A00&per_page=250](https://api.safecast.org/measurements.json?since=2022-09-08+00%3A00&until=2022-09-09+00%3A00&per_page=250)

Would that work for you?

Regards,
Rob

**Comment by adanecito** [comment]
YES!!!! Many thanks!
-Tony

**Comment by robouden** [comment]
Happy that worked for you.

Regards,
rob


Regards,
Rob Oudendijk Yuka Hayashi
http://yr-design.biz   http://oudendijk.biz  http://about.me/robouden
tel +81 80-22605966  Skype: robouden  Facebook:robouden
<http://on.fb.me/QeKw2P> twitter:robouden <http://bit.ly/RAiSTC>

**Comment by adanecito** [comment]
I am Tony Anecito, http://www.myuniportal.com, 1- 208 284-8960 I do not have skype or facebook. I do have twitter but do not use it.
My android apps (5) are on google play. Real Moon 3D , Real Mars 3D, Wildfire Locations Now, Air Quality Now, Earthquake Locations Now

I had one more question. I hate to download so much data for a day. Is there some way only to get records for the same user/device only once and for it to be the latest? I want to be a good person in regards to the resources used like bandwidth, CPU. I can do additional filtering at the Android smartphone.

Again many thanks,
-Tony

**Reply by robouden** [reply]
Tony,

If you have a look at the code for realtime.safecast.org on GitHub, you can see how we pull the data from the API for "real-time" sensors. Every 15 mites we pull the latest data from the devices registered.

Regard,
Rob

**Comment by adanecito** [comment]
Perfect. You are a lifesaver!!
Regard,
-Tony

**Comment by adanecito** [comment]
I ran into an interesting issue with the data. In the last month there has been high CPM over 1000. I am guessing it was just one device. The capturedAt was listed in 2080 or 2050 and the ids were strange. I ended up filtering those out. Just thought you should know.

Regards,
-Tony

**Comment by robouden** [comment]
Tony,

Thanks for letting us know.

Regards,
Rob Oudendijk

Regards,
Rob Oudendijk Yuka Hayashi
http://yr-design.biz   http://oudendijk.biz  http://about.me/robouden
tel +81 80-22605966  Skype: robouden  Facebook:robouden
<http://on.fb.me/QeKw2P> twitter:robouden <http://bit.ly/RAiSTC>

**Comment by adanecito** [comment]
Look for records with device_id = 0

**Comment by laujonat** [comment]
Just a heads up that all API resources at `api.safecast.org` are currently down

**Comment by robouden** [comment]
Jonathan,

Thanks for doing that.

Regards
Rob

**Comment by robouden** [comment]
Rolled back the update and the API is back up.
https://api.safecast.org/

Rob
